src/QsCompiler/QirGeneration/Subtransformations/ExpressionKindTransformation.cs

```diff
@@ -2260,7 +2260,7 @@ public override ResolvedExpressionKind OnRightShift(TypedExpression lhsEx, Typed
             IValue value;
             if (exType.Resolution.IsInt)
             {
-                var res = this.SharedState.CurrentBuilder.ArithmeticShiftRight(lhs.Value, rhs.Value);
+                var res = this.SharedState.CurrentBuilder.LogicalShiftRight(lhs.Value, rhs.Value);
```

Comment:
I see above that for the left shift there is only the direct use of left shift, which I believe should give us the right behavior. Could you confirm?

Reply:
Yes, that's correct. The shift left instruction does not have a variant, only `shl` that we are already using.